src/coreclr/src/debug/daccess/request.cpp

```diff
@@ -4500,3 +4500,180 @@ HRESULT ClrDataAccess::GetMethodsWithProfilerModifiedIL(CLRDATA_ADDRESS mod, CLR
 
     return hr;
 }
+
+HRESULT ClrDataAccess::GetNumberGenerations(unsigned int *pGenerations)
+{
+    if (pGenerations == NULL)
+    {
+        return E_INVALIDARG;
+    }
+
+    SOSDacEnter();
+
+    *pGenerations = (unsigned int)(g_gcDacGlobals->total_generation_count);
+
+    SOSDacLeave();
+    return S_OK;
+}
+
+HRESULT ClrDataAccess::GetGenerationTable(unsigned int cGenerations, struct DacpGenerationData *pGenerationData, unsigned int *pNeeded)
```

Comment:
>unsigned int *pNeeded [](start = 113, length = 21)

is the only purpose of this arg is for verification on the sos side?

Reply:
It's so you can request the number of generations and dynamically allocated an array of the right size

```
unsigned int count;
if (SUCCEEDED(pSos->GetGenerationTable(0, NULL, &count)
{
    DacpGenerationData *pGenerationData = new DacpGenerationData[count];
    pSos->GetGenerationTable(count, pGenerationData, NULL);
}
```

Reply:
I see. I didn't know it was going to be called this way. makes sense. thanks!